spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,497 @@
+    img.change_orientation('RPI')
+    path_ref = os.path.join(__data_dir__, 'PAM50_normalized_metrics')
+    if arguments.o is not None:
+        fname_out = arguments.o
+    else:
+        path, file_name, ext = extract_fname(get_absolute_path(arguments.i))
+        fname_out = os.path.join(path, file_name + '_compression_metrics' + ext)
+    fname_metrics = get_absolute_path(arguments.i)
+    metric = 'MEAN(' + arguments.metric + ')'  # Adjust for csv file columns name
+    # Fetch metrics of subject
+    df_metrics = csv2dataFrame(fname_metrics, metric)
+    # Get vertebral level corresponding to the slice with the compression
+    slice_thickness = get_slice_thickness(img)
+    slice_compressed = get_compressed_slice(img, verbose)
+
+    compressed_levels_dict = get_verterbral_level_from_slice(slice_compressed, df_metrics)
+    # Get vertebral level above and below the compression
+    upper_level, lower_level = get_up_lw_levels(compressed_levels_dict.keys(), df_metrics, metric)
+    # Initialize variables if normalization with
+    if arguments.i_PAM50:
+        fname_metrics_PAM50 = get_absolute_path(arguments.i_PAM50)
+        sex = arguments.sex
+        age = arguments.age
+        if age:
+            if any(n < 0 for n in age):
+                parser.error('Age range needs to be positive, {} was specified'.format(age))
+            # Put age range in order
+            else:
+                age.sort()
+        if sex or age:
+            if not os.path.isfile(get_absolute_path(os.path.join(path_ref, arguments.file_participants))):
+                raise FileNotFoundError('participants.tsv file must exists to select sex or age.')
+            else:
+                fname_partcipants = get_absolute_path(os.path.join(path_ref, arguments.file_participants))
+                list_HC = select_HC(fname_partcipants, sex, age)
+        else:
+            list_HC = None
+        # Select healthy controls based on sex and/or age range
+
+        # Get PAM50 slice thickness
+        fname_PAM50 = os.path.join(__data_dir__, 'PAM50', 'template', 'PAM50_t2.nii.gz')
+        img_pam50 = Image(fname_PAM50)
+        img_pam50.change_orientation('RPI')
+        slice_thickness_PAM50 = get_slice_thickness(img_pam50)
+        # Fetch metrics of PAM50 template
+        df_metrics_PAM50 = csv2dataFrame(fname_metrics_PAM50, metric)
+
+        # Get slices corresponding in PAM50 space
+        compressed_levels_dict = get_slices_in_PAM50(compressed_levels_dict, df_metrics, df_metrics_PAM50)
+
+        # Get data from healthy control and average them
+        df_avg_HC = average_hc(path_ref, metric, list_HC)
+
+    # Loop through all compressed levels (compute one MSCC per compressed level)
+    for level in compressed_levels_dict.keys():
+        # Get metric of patient with compression
+        if arguments.i_PAM50:
+            ap, slices_avg = average_compression_PAM50(slice_thickness, slice_thickness_PAM50, metric, df_metrics_PAM50,
```

Comment:
Since we allowed to normalize also other metrics (`['diameter_AP', 'area', 'diameter_RL', 'eccentricity', 'solidity']`) using the`-metric` input argument, we should probably rename the `ap` variable. The `ap` referred to  diameter_AP. But now, it can contain any of the listed metrics.

Reply:
done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4003/commits/634e2f3a7c105d4c62931df3845878bde9c525c2